src/core/funcbloc.cpp

```diff
@@ -636,11 +636,30 @@ void CFunctionBlock::traceInputEvent(TEventID paEIID){
   }
 }
 
-void CFunctionBlock::traceOutputEvent(TEventID paEOID){
+void CFunctionBlock::traceOutputEvent(TEventID paEOID, CEventChainExecutionThread * const paECET){
   if(auto& tracer = getResource()->getTracer(); tracer.isEnabled()){
+ 
+    // don't trace events that are not connected since they don't increase the event counter
+    if(getEOConUnchecked(static_cast<TPortId>(paEOID))->getDestinationList().size() == 0) {
+      return;
+    }
+
+    std::vector<std::string> outputs(getFBInterfaceSpec().mNumDOs);
```

Comment:
with your change data outputs are now logged twice. is your change needed? can't you use the existing data output logging?

Reply:
I separated the two tracing methods: one for the classic one, and a new one for replay debugging. The replay debugging one changes the interface of the `sendOutputEvent` so I added a CMake variable and a `define` to separate both properly. For now, the replay debugging one still traces as much as the regular one,  but eventually the I will remove the rest of the traces for that configuration.

I had to move some files around to leave it cleaner, but basically I added some folders for it. The `internal` and `reader` folders are used only for testing and for the replay debugging device.